logging/common/src/main/java/io/helidon/logging/common/BaseMdcProvider.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2025 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.logging.common;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+import io.helidon.logging.common.spi.MdcProvider;
+
+/**
+ * Implements {@link io.helidon.logging.common.spi.MdcProvider} using an internal thread local data structure.
+ */
+public class BaseMdcProvider implements MdcProvider {
+
+    /*
+    Each instance of this class gets its own thread local map rather than usual practice of declaring thread local fields as
+    static. That's because each instance should represent a different provider type and there should be exactly one instance
+    for each provider type. This way each different provider type has its own thread local map in case, for some reason,
+    the maps for different providers need to be different.
+     */
+    private final ThreadLocal<Map<String, Supplier<String>>> mdcSettings = ThreadLocal.withInitial(HashMap::new);
+
+    /**
+     * Creates a new instance.
+     */
+    public BaseMdcProvider() {
+    }
+
+    @Override
+    public void put(String key, Supplier<String> valueSupplier) {
+        mdcSettings.get().put(key, valueSupplier);
+    }
+
+    @Override
+    public void put(String key, String value) {
+        mdcSettings.get().put(key, value::toString);
+    }
+
+    @Override
+    public void remove(String key) {
+        mdcSettings.get().remove(key);
+    }
+
+    @Override
+    public void clear() {
+        mdcSettings.get().clear();
+    }
+
+    @Override
+    public String get(String key) {
+        // Normally Helidon APIs do not return nulls, but this is done here for backwards compatibility with
+        // earlier implementations.
+        return mdcSettings.get().containsKey(key) ? mdcSettings.get().get(key).get() : null;
+    }
+
+    /**
+     * Returns the current key/supplier MDC settings.
+     *
+     * @return copy of the current key/supplier map containing the current MDC settings
+     */
+    public Map<String, Supplier<String>> settings() {
```

Comment:
These methods should not be public - they should only be used within the same package (i.e. `protected` here).

Reply:
No longer using a common base provider.